cristin-organization/src/main/java/no/unit/nva/cristin/organization/common/client/version20230526/CristinOrgApiClient20230526.java

```diff
@@ -0,0 +1,140 @@
+package no.unit.nva.cristin.organization.common.client.version20230526;
+
+import static java.util.Arrays.asList;
+import static no.unit.nva.HttpClientProvider.defaultHttpClient;
+import static no.unit.nva.cristin.model.Constants.CRISTIN_API_URL;
+import static no.unit.nva.cristin.model.Constants.CRISTIN_PER_PAGE_PARAM;
+import static no.unit.nva.cristin.model.Constants.NONE;
+import static no.unit.nva.cristin.model.Constants.ORGANIZATION_PATH;
+import static no.unit.nva.cristin.model.Constants.PARENT_UNIT_ID;
+import static no.unit.nva.cristin.model.Constants.UNITS_PATH;
+import static no.unit.nva.cristin.model.JsonPropertyNames.DEPTH;
+import static no.unit.nva.cristin.model.JsonPropertyNames.IDENTIFIER;
+import static no.unit.nva.cristin.organization.common.QueryParamConverter.translateToCristinApi;
+import static no.unit.nva.model.Organization.ORGANIZATION_CONTEXT;
+import static no.unit.nva.utils.UriUtils.createCristinQueryUri;
+import static no.unit.nva.utils.UriUtils.getNvaApiUri;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import no.unit.nva.cristin.common.client.ApiClient;
+import no.unit.nva.cristin.common.client.FetchApiClient;
+import no.unit.nva.cristin.common.client.QueryApiClient;
+import no.unit.nva.cristin.model.SearchResponse;
+import no.unit.nva.cristin.organization.dto.version20230526.mapper.OrganizationFromUnitMapper;
+import no.unit.nva.cristin.organization.dto.version20230526.UnitDto;
+import no.unit.nva.model.Organization;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.BadGatewayException;
+import nva.commons.core.paths.UriWrapper;
+
+public class CristinOrgApiClient20230526 extends ApiClient
```

Comment:
class is easy to read 👍

Reply:
i am considering splitting query and fetch into their own separate clients, because i dont think we need shared methods after all. The goal was to either doing a search with a implicit full query for each, but then it might have gotten very slow and verbose. so im thinking now that client does a separate search and then fetches one at a time for their requested search result